meta-lmp-bsp/recipes-bsp/u-boot/u-boot-ostree-scr-fit/imx6ullevk/boot.cmd

```diff
@@ -10,18 +10,149 @@ setenv devnum ${mmcdev}
+fi
+
+if test "${fiovb.is_secondary_boot}" = "0"; then
+	if test -n "${board_is_closed}"; then
+		# FIO bootcount, upgrade_available and rollback initialization
+		mmc rescan
+		fiovb init ${devnum}
+
+		fiovb read_pvalue bootcount 4
+		if test ! $? -eq 0; then fiovb write_pvalue bootcount 0; fi
+
+		fiovb read_pvalue rollback 4
+		if test ! $? -eq 0; then fiovb write_pvalue rollback 0; fi
+
+		fiovb read_pvalue upgrade_available 4
+		if test ! $? -eq 0; then fiovb write_pvalue upgrade_available 0; fi
+
+		fiovb read_pvalue bootupgrade_available 4
+		if test ! $? -eq 0; then fiovb write_pvalue bootupgrade_available 0; fi
+	fi
+
+	# Only update bootcount when upgrade_available is set
+	if test "${fiovb.upgrade_available}" = "1"; then
+		setexpr bootcount ${fiovb.bootcount} + 1;
+		if test -n "${board_is_closed}"; then
+			fiovb write_pvalue bootcount ${bootcount};
+		fi;
+
+		# If bootcount is above bootlimit, trigger rollback
+		# if not - try to boot secondary boot path for boot firmware validation
+		if test ${bootcount} -gt ${bootlimit}; then
+			echo "FIO: doing rollback..."
+			if test -n "${board_is_closed}"; then
+				fiovb write_pvalue rollback 1;
+				fiovb write_pvalue upgrade_available 0;
+				fiovb write_pvalue bootupgrade_available 0;
+			else
+				setenv rollback 1;
+				setenv upgrade_available 0;
+				setenv bootupgrade_available 0;
+			fi;
+		else
+			if test "${fiovb.bootupgrade_available}" = "1"; then
+				imx_secondary_boot 1;
+				echo "FIO: updating secondary boot images from ${ostree_root} ..."
+
+				setenv image_path "${ostree_root}/usr/lib/firmware/${bootloader_image}"
+				echo "FIO: Writing ${image_path} ..."
+				setenv start_blk "${bootloader_s}"
+				run load_image
+				run update_image
+
+				setenv image_path "${ostree_root}/usr/lib/firmware/${bootloader2_image}"
+				echo "FIO: Writing ${image_path} ..."
+				setenv start_blk "${bootloader2_s}"
+				run load_image
+				run update_image
+
+				run saveenv_mmc
+				echo "FIO: doing warm reset to boot into secondary boot path..."
+				reset -w
+			fi;
+		fi
+	fi
+
+	# That means that we've validated already boot image over secondary path
+	if test "${fiovb.bootupgrade_available}" = "1" && test "${fiovb.upgrade_available}" = "0"; then
+		echo "FIO: update primary boot path with validated images ..."
+
+		setenv image_path "${ostree_root}/usr/lib/firmware/${bootloader_image}"
+		echo "FIO: Writing ${image_path} ..."
+		setenv start_blk "${bootloader}"
+		run load_image
+		run update_image
+
+		setenv image_path "${ostree_root}/usr/lib/firmware/${bootloader2_image}"
+		echo "FIO: Writing ${image_path} ..."
+		setenv start_blk "${bootloader2}"
+		run load_image
+		run update_image
+
+		run bootcmd_bootenv
+		if test -n "${board_is_closed}"; then
+			fiovb write_pvalue bootupgrade_available 0;
+			fiovb write_pvalue bootfirmware_version ${bootfirmware_version};
+		else
+			setenv bootupgrade_available 0;
+		fi
+
+		run saveenv_mmc
+		echo "FIO: reset ..."
```

Comment:
Do we need `imx_secondary_boot 0;` here similar to the `apalis-imx6` boot.cmd?

Reply:
not really, will drop that

Reply:
well, we don't need this as `reset` leads to POR, so we really don't care about state of SECONDARY_BOOT bit